msg/versioned/BatteryStatus.msg

```diff
@@ -1,74 +1,77 @@
+uint16 manufacture_date # Manufacture date, part of serial number of the battery pack. Formatted as: Day + Month×32 + (Year–1980)×512
+uint16 state_of_health # State of health. FullChargeCapacity/DesignCapacity [%] [@range 0, 100].
+uint16 max_error # Max error, expected margin of error in the state-of-charge calculation [%] [@range 1, 100]
+uint8 id # ID number of a battery. Should be unique and consistent for the lifetime of a vehicle. 1-indexed.
+uint16 interface_error # Interface error counter
 
-bool is_powering_off				# Power off event imminent indication, false if unknown
-bool is_required				# Set if the battery is explicitly required before arming
+float32[14] voltage_cell_v # Battery individual cell voltages [V] [@invalid 0]
+float32 max_cell_voltage_delta # Max difference between individual cell voltages
 
+bool is_powering_off # Power off event imminent indication, false if unknown
+bool is_required # Set if the battery is explicitly required before arming
 
-uint8 WARNING_NONE = 0			# no battery low voltage warning active
-uint8 WARNING_LOW = 1			# warning of low voltage
-uint8 WARNING_CRITICAL = 2		# critical voltage, return / abort immediately
-uint8 WARNING_EMERGENCY = 3		# immediate landing required
-uint8 WARNING_FAILED = 4		# the battery has failed completely
-uint8 STATE_UNHEALTHY = 6 		# Battery is diagnosed to be defective or an error occurred, usage is discouraged / prohibited. Possible causes (faults) are listed in faults field.
-uint8 STATE_CHARGING = 7 		# Battery is charging
+uint8 warning # Current battery warning [@enum WARNING]
+uint8 WARNING_NONE = 0 # No battery low voltage warning active
+uint8 WARNING_LOW = 1 # Low voltage warning
+uint8 WARNING_CRITICAL = 2 # Critical voltage, return / abort immediately
+uint8 WARNING_EMERGENCY = 3 # Immediate landing required
+uint8 WARNING_FAILED = 4 # Battery has failed completely
 
-uint8 FAULT_DEEP_DISCHARGE = 0 		# Battery has deep discharged
-uint8 FAULT_SPIKES = 1 			# Voltage spikes
-uint8 FAULT_CELL_FAIL= 2 		# One or more cells have failed
-uint8 FAULT_OVER_CURRENT = 3 		# Over-current
-uint8 FAULT_OVER_TEMPERATURE = 4 	# Over-temperature
-uint8 FAULT_UNDER_TEMPERATURE = 5 	# Under-temperature fault
-uint8 FAULT_INCOMPATIBLE_VOLTAGE = 6 	# Vehicle voltage is not compatible with this battery (batteries on same power rail should have similar voltage).
-uint8 FAULT_INCOMPATIBLE_FIRMWARE = 7 	# Battery firmware is not compatible with current autopilot firmware
-uint8 FAULT_INCOMPATIBLE_MODEL = 8 	# Battery model is not supported by the system
-uint8 FAULT_HARDWARE_FAILURE = 9 	# hardware problem
-uint8 FAULT_FAILED_TO_ARM = 10 		# Battery had a problem while arming
-uint8 FAULT_COUNT = 11 			# Counter - keep it as last element!
+uint8 STATE_UNHEALTHY = 6 # Battery is diagnosed to be defective or an error occurred, usage is discouraged / prohibited. Possible causes (faults) are listed in faults field.
+uint8 STATE_CHARGING = 7 # Battery is charging
```

Comment:
These two values don't seem to be associated with a field.

If they are allowed values of warning, we could either rename to have the same prefix OR we could do

> uint8 warning # Current battery warning [@enum WARNING STATE]

Reply:
These are basically "charge states": https://mavlink.io/en/messages/common.html#MAV_BATTERY_CHARGE_STATE Historically the field name is warning which I might also propose to change.

Reply:
OK, so this is a case we do need to resolve for the docs approach. We either need to give them the common prefix or allow multiple prefixes as in the first comment. 
I think allowing multiple prefixes as per https://github.com/PX4/PX4-Autopilot/pull/24789/files#r2078787661 is the most flexible approach, because it can always be reverted back to a single prefix.